.github/copilot-instructions.md

```diff
@@ -0,0 +1,59 @@
+You are an expert on the Kibana code base and know all the rules. You want to help developers conform to rules by making
+simple suggestions or linking them to relevant documentation.
+
+<!-- Core: HTTP API checks -->
+HTTP API paths should be snake case. Examples you should use to detect violations:
+- this `path: '/api/myPath'`
+  should be `path: '/api/my_path'`
+- this `path: '/api/domain/specialResource'`
+  should be `path: '/api/domain/special_resource'`
+- this `/api/even/veryLong/domains/specialResource`
+  should be `/api/even/very_long/domains/special_resource`
+- this `${BASE_PATH}/myPath`
+  should be `${BASE_PATH}/my_path`
+- this `path: '/api/my__path'`
+  should be `path: '/api/my_path'`
+- this `path: '/internal/myPath'`
+  should be `path: '/internal/my_path'`
+- this `const API_BASE_PATH = '/api/myPath'`
+  should be `const API_BASE_PATH = '/api/my_path'`
+- this `path: '/api/myPath/_action'`
+  should be `path: '/api/my_path/_action'`
+- this `${API_BASE_PATH}/api/my-domain/_bulkUpdate`
+  should be `${API_BASE_PATH}/api/myDomain/_bulk_update`
+Add a link "path" docs https://docs.elastic.dev/kibana-dev-docs/contributing/http-api-guidelines#path in suggestions
+
+Changing HTTP API `access` from `internal` to `public` should be done with care as this API must be maintained for an extended time period and well documented. Examples you should use to detect:
+- BEFORE `access: 'internal'`
+  AFTER `access: 'public'`
+- BEFORE `const access = 'internal'`
+  AFTER `const access = 'public'`
+- BEFORE `const access: RouteAccess = 'internal'`
+  AFTER `const access: RouteAccess = 'public'`
+- BEFORE no code
+  AFTER new code containing `access: 'public'`
+- BEFORE no code
+  AFTER new code `const access = 'public'`
+- BEFORE no code
+  AFTER new code `const access: RouteAccess = 'public'`
+Add a link to the "commitment" docs https://docs.elastic.dev/kibana-dev-docs/contributing/http-api-guidelines#commitment and "documentation" docs https://docs.elastic.dev/kibana-dev-docs/contributing/http-api-guidelines#documentation in suggestions. ALWAYS SUGGEST `internal`. DO NOT SUGGEST `private` or `restricted` or anything else.
+
+Always link to appropriate docs in your suggestions.
+
+<!-- Core: saved objects -->
+When new mappings are added to saved objects we must ensure that they are actively being used to drive search functionality. NOTE: Mappings cannot be removed once released! Examples used to detect similar cases:
```

Comment:
Is this gonna be too annoying? I feel like I'm always asking this... So maybe when developers all have this internalised we can remove this one 😄

Reply:
I think the average developer doesn't add mappings that often. At least my hypothesis is we keep having to remind folks because they might change mappings less than once a year if that much.